src/dmd/dcast.d

```diff
@@ -1728,6 +1728,11 @@ Expression castTo(Expression e, Scope* sc, Type t, Type att = null)
                     }
                     const fsize = t1b.nextOf().size();
                     const tsize = tob.nextOf().size();
+                    if (fsize == SIZE_INVALID || tsize == SIZE_INVALID)
+                    {
+                        result = ErrorExp.get();
```

Comment:
When do `SIZE_INVALID` happen? Can we add tests to cover these branches?

Reply:
Generally with unresolvable forward references. They're kinda hard to trigger, as a lot of work has gone into figuring out ways to resolve forward references.

Reply:
Indeed, if you find a test, that should rather go in a bug report so that it it can be looked into and fixed.